crates/bitwarden/src/vault/cipher/cipher.rs

```diff
@@ -185,3 +185,87 @@ impl Decryptable<CipherView> for Cipher {
         })
     }
 }
+
+impl Cipher {
+    fn get_decrypted_subtitle(
+        &self,
+        enc: &EncryptionSettings,
+        org_id: &Option<Uuid>,
+    ) -> Result<String> {
+        Ok(match self.r#type {
+            CipherType::Login => {
+                let Some(login) = &self.login else { return Ok(String::new()) };
+                login.username.decrypt(enc, org_id).unwrap_or_default()
+            }
+            CipherType::SecureNote => String::new(),
+            CipherType::Card => {
+                let Some(card) = &self.card else { return Ok(String::new()) };
+                let mut sub_title = String::new();
+
+                if let Some(brand) = &card.brand {
+                    sub_title.push_str(&brand.decrypt(enc, org_id)?);
+                }
+
+                if let Some(number) = &card.number {
+                    let number = number.decrypt(enc, org_id)?;
+                    let number_len = number.len();
+                    if number_len > 4 {
+                        if !sub_title.is_empty() {
+                            sub_title.push_str(", ");
+                        }
+
+                        // On AMEX cards we show 5 digits instead of 4
+                        let digit_count = match &number[0..2] {
+                            "34" | "37" => 5,
+                            _ => 4,
+                        };
```

Comment:
Should we use the card type instead of number for this? Not sure what the TS implementation uses.

Reply:
The clients seem to use the number the same way, not sure if there's a reason for that (Maybe in case a user mistypes the card?)

https://github.com/bitwarden/clients/blob/fb74c2d6eeef9d974036e58dbba5dfe30e992dbe/libs/common/src/vault/models/view/card.view.ts#L58-L60